Sources/SwiftFormat/Rules/DontRepeatTypeInStaticProperties.swift

```diff
@@ -22,69 +22,26 @@ import SwiftSyntax
-    diagnoseStaticMembers(node.memberBlock.members, endingWith: node.name.text)
-    return .skipChildren
-  }
-
-  public override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
-    diagnoseStaticMembers(node.memberBlock.members, endingWith: node.name.text)
-    return .skipChildren
-  }
-
-  public override func visit(_ node: ExtensionDeclSyntax) -> SyntaxVisitorContinueKind {
-    let members = node.memberBlock.members
-
-    switch Syntax(node.extendedType).as(SyntaxEnum.self) {
-    case .identifierType(let simpleType):
-      diagnoseStaticMembers(members, endingWith: simpleType.name.text)
-    case .memberType(let memberType):
-      // We don't need to drill recursively into this structure because types with more than two
-      // components are constructed left-heavy; that is, `A.B.C.D` is structured as `((A.B).C).D`,
-      // and the final component of the top type is what we want.
-      diagnoseStaticMembers(members, endingWith: memberType.name.text)
-    default:
-      // Do nothing for non-nominal types. If Swift adds support for extensions on non-nominals,
-      // we'll need to update this if we need to support some subset of those.
-      break
+  /// Visits the static/class properties and diagnoses any where the name has the containing
+  /// type name (excluding possible namespace prefixes, like `NS` or `UI`) as a suffix.
+  public override func visit(_ node: VariableDeclSyntax) -> SyntaxVisitorContinueKind {
+    guard node.modifiers.contains(anyOf: [.class, .static]), let typeName = Syntax(node).typeName else {
+      return .visitChildren
     }
 
-    return .skipChildren
-  }
-
-  /// Iterates over the static/class properties in the given member list and diagnoses any where the
-  /// name has the containing type name (excluding possible namespace prefixes, like `NS` or `UI`)
-  /// as a suffix.
-  private func diagnoseStaticMembers(_ members: MemberBlockItemListSyntax, endingWith typeName: String) {
-    for member in members {
-      guard
-        let varDecl = member.decl.as(VariableDeclSyntax.self),
-        varDecl.modifiers.contains(anyOf: [.class, .static])
-      else { continue }
-
-      let bareTypeName = removingPossibleNamespacePrefix(from: typeName)
-
-      for binding in varDecl.bindings {
-        guard let identifierPattern = binding.pattern.as(IdentifierPatternSyntax.self) else {
-          continue
-        }
+    let bareTypeName = removingPossibleNamespacePrefix(from: typeName)
+    for binding in node.bindings {
+      guard let identifierPattern = binding.pattern.as(IdentifierPatternSyntax.self) else {
+        continue
+      }
 
-        let varName = identifierPattern.identifier.text
-        if varName.contains(bareTypeName) {
-          diagnose(.removeTypeFromName(name: varName, type: bareTypeName), on: identifierPattern)
-        }
+      let varName = identifierPattern.identifier.text
+      if varName.contains(bareTypeName) {
```

Comment:
Now that I'm getting another look at this, I wonder if we should update this to be `hasSuffix`, since the diagnostic explicitly talks about suffixes.

Reply:
Good question...  we wouldn't catch plurals (which we don't mention or have tests for), but there are likely false positives with the current logic too.
```
public actor Cookie {
  // cases that are caught with .contains but not with .hasSuffix
  static let chocolateChipCookies: Cookie
  static let rawCookieDough: Cookie
}
```